frontend/src/scenes/organization/ConfirmOrganization/confirmOrganizationLogic.ts

```diff
@@ -50,7 +50,15 @@ export const confirmOrganizationLogic = kea<confirmOrganizationLogicType>([
                     const response = await api.create('api/social_signup/', {
                         ...formValues,
                     })
-                    location.href = response.success_url || '/'
+
+                    const nextUrl = router.values.searchParams.get('next')
+
+                    const successUrl = response.success_url || '/'
+
+                    const redirectUrl = new URL(successUrl)
+                    redirectUrl.searchParams.set('next', nextUrl || '')
+
+                    location.href = redirectUrl.toString()
```

Comment:
This is passing through so it'll appear in onboarding, would onboarding handle this next at the end of it?